pkg/resource/deploy/source_eval.go

```diff
@@ -457,12 +493,24 @@ func (rm *resmon) getProvider(pkg tokens.Package, rawProviderRef string) (plugin
 	return provider, nil
 }
 
+func (rm *resmon) parseProviderRequest(pkg tokens.Package, version string) providers.ProviderRequest {
```

Comment:
Just wondering here, in cases where we can't parse the version, should this just fail instead?  As is, does this mean you'll just get the "latest" plugin when you pass an invalid version?

Reply:
There are two reasons that we could fail to parse a semver:

1. The version is the empty string, i.e. the language host didn't send a version. This fails to parse, and we continue on as we did before (you get the "latest" plugin, or you get a default plugin if one exist in the default plugin map)
2. The version is random garbage

I think it makes sense to split these two up and to continue on 1) and fail on 2).

Reply:
> I think it makes sense to split these two up and to continue on 1) and fail on 2).

Yes, I agree.